sklearn/model_selection/tests/test_split.py

```diff
@@ -1538,3 +1539,84 @@ def test_leave_p_out_empty_trainset():
             ValueError,
             match='p=2 must be strictly less than the number of samples=2'):
         next(cv.split(X, y, groups=[1, 2]))
+
+
+def test_binnedstratifiedkfold_error():
+    with pytest.raises(ValueError, match='Need at least two bins'):
+        BinnedStratifiedKFold(n_bins=1)
+
+
+def test_binnedstratifiedkfold_balance():
+    rng = np.random.RandomState(0)
+    for _ in range(10):
+        n_splits = 2 + int(10 * rng.rand())
+        X, y = make_regression(noise=10, random_state=rng)
+        train_sizes = []
+        test_sizes = []
+
+        bskf = BinnedStratifiedKFold(n_splits=n_splits, shuffle=False,
+                                     random_state=None)
```

Comment:
maybe void random_state=None in tests?

Reply:
shuffle is false anyway but yes, this is redundant.